### ISSUE TITLE
alternate key code when using shift

### ISSUE THREAD

**Issue by mirsella** [florisboard/florisboard#145]
i think it would be useful if in the layout.json files (app/src/main/assets/ime/text/characters/*.json) we could specify an alternate key code when caps is enabled.
especially since in the roadmap you want to add an option to easily import custom layout in the app.

gboard has this feature at least on azerty : the `'` key become a `?` when pressing shift.

maybe something like this in the layout.json : 
```json
      { "code": 39, "label": "'", "shift": {
        "code": 34, "label": "\""}
      }
```
so if key.shift then use it else key.toUpperCase()

thanks !

**Comment by patrickgold** [comment]
Thanks for your detailed issue! The way you described the implementation is actually the simplest yet best implementation and I will probably add it exactly this way. :)

**Comment by patrickgold** [comment]
Above PR adds an option for keys to define an override shift key data, see the provided demo in the PR for how it looks. I have modified Dvorak and Colemak layout to take advantage of this new feature. Will be released in v0.3.9. Thanks again for your suggestion!
